xbmc/GUITextureJobManager.cpp

```diff
@@ -0,0 +1,107 @@
+/*
+ *  Copyright (C) 2024 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "GUITextureJobManager.h"
+
+#include "ServiceBroker.h"
+#include "settings/AdvancedSettings.h"
+#include "settings/SettingsComponent.h"
+#include "windowing/WinSystem.h"
+
+using namespace std::chrono_literals;
+using namespace KODI::GUILIB;
+
+CGUITextureLoaderThread::CGUITextureLoaderThread(CGUITextureJobManager* manager,
+                                                 unsigned int threadID)
+  : CThread("TextureLoader"), m_threadID(threadID), m_manager(manager)
+{
+  Create();
+  SetPriority(ThreadPriority::LOWEST);
+}
+
+void CGUITextureLoaderThread::OnStartup()
+{
+  if (CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_guiAsyncTextureUpload)
+    m_hasContext = CServiceBroker::GetWinSystem()->BindSecondaryGPUContext(m_threadID);
+}
+
+void CGUITextureLoaderThread::Process()
+{
+  while (!m_bStop)
+  {
+    CImageLoader* image = m_manager->GetNextImage();
+    if (image)
+    {
+      image->DoWork(m_hasContext);
+      image->m_callback->OnLoadComplete(image);
+    }
+    else
+    {
+      // sleep for a frame, assuming 60fps
+      CThread::Sleep(16ms);
+    }
+  }
```

Comment:
I'm not a fan of loops with sleeps. How about working with a condition variable instead? E.g. like this: https://github.com/neo1973/xbmc/commit/04122748df7d2b5dd0876ab73c4b436a142c5405